### PR TITLE
Update Terraform behavior to properly forward notification server traffic

### PR DESCRIPTION
**Pull request by soulwa**
# Purpose

###### A brief description of the purpose of the PR's changes for the project.

<br>

Update Terraform to properly forward data requests to the webserver. 

The webserver container in AWS deploys both the GraphQL API and the notifications server (on different ports). To forward them, we create rules based on the forwarding domain. 

So — `notifs.searchneu.com` should be forwarded to the notification server, and `api.searchneu.com` should be forwarded to the GraphQL API. 

We were having an issue with the staging notifications — the notifications were being forwarded to the GraphQL API.

The issue was with how our domains were declared in Terraform. @hankewyczz made changes to Terraform during the migration to ensure that `notifs.searchneu.com` was getting a Cloudflare cert (before that, this wasn't configured in Terraform). However, this had the side-effect of messing up the rule forwarding traffic. 

# Tickets

###### A link to any tickets this PR is associated with on Trello.

- N/A

# Contributors

###### Anyone who contributed to this PR for future reference.

- @hankewyczz 
- @soulwa 

<br>
@sandboxnu/searchneu

### CODE REVIEW

File: infrastructure/terraform/modules/course-catalog-api/variables.tf [hunk 1]
```diff
@@ -32,8 +32,12 @@ variable "alb_dns_name" {
   description = "alb dns name to setup cname to"
 }
 
-variable "domains" {
-  description = "domains to put this env on. ALB needs to already have certs for it"
+variable "api_domain" {
+  description = "domain for the API server"
```
Comment (hankewyczz) [inline infrastructure/terraform/modules/course-catalog-api/variables.tf:36]:
```suggestion
  description = "domain for the API server. ALB needs to already have certs for it"
```

File: infrastructure/terraform/modules/course-catalog-api/variables.tf [hunk 2]
```diff
@@ -32,8 +32,12 @@ variable "alb_dns_name" {
...
+}
+
+variable "notifs_domain" {
+  description = "domain for the notification server"
```
Comment (hankewyczz) [inline infrastructure/terraform/modules/course-catalog-api/variables.tf:40]:
```suggestion
  description = "domain for the notification server. ALB needs to already have certs for it"
```

**Review by hankewyczz — COMMENTED** [review]
minor tweak

**Comment by soulwa** [comment]
should I look at changing the cloudflare cname resource? wasn't sure if we want that to be cleaner than creating an adhoc array of the domain urls.

**Comment by hankewyczz** [comment]
> should I look at changing the cloudflare cname resource? wasn't sure if we want that to be cleaner than creating an adhoc array of the domain urls.

sure, but don't spend too much time on it — this is a small-enough bit of code that I'd ultimately be OK with keeping it as-is (though I would prefer a cleaner solution).

**Review by Lucas-Dunker — APPROVED** [review]
This isn't my area of expertise, but I'd love to hear how these changes work at standup!

**Comment by soulwa** [comment]
changed `count` to the cleaner `for_each` metavariable, all tests are now passing after merging `master` into this branch, so should be all set.

**Review by Lucas-Dunker — APPROVED** [review]
Looks good!

**Review by sebwittr — APPROVED** [review]
For the most part looks good to me. I'm not too familiar with terraform, but the majority of the changes made sense.

**Review by pranavphadke1 — APPROVED** [review]

**Pull request merged**